airbyte-integrations/connectors/source-shopify/source_shopify/http_request.py

```diff
@@ -35,10 +38,13 @@ def interpret_response(self, response: Optional[Union[requests.Response, Excepti
             )
         elif isinstance(response, requests.Response):
             if response.ok:
-                return ErrorResolution(ResponseAction.IGNORE, None, None)
+                return _NO_ERROR_RESOLUTION
+
+            if response.status_code == 429 or response.status_code >= 500:
```

Comment:
This behavior was part of the HttpStream which was used for bulk job creation requests. Hence, I moved the logic here so that it applies to all the HTTP requests performed by HttpClient